### PR TITLE
Baseline - Refactor AWS check stack #373

### LINKED ISSUE #373: Baseline - Refactor AWS check stack
- [ ] Create README.md
- [ ] Add outputs.tf
- [ ] Add variables.tf
- [ ] Add providers.tf
- [ ] Add smart_folder.tf
- [ ] Categories polices
- [ ] Add guardrail (category) links
- [ ] Add mod links
- [ ] Refactor to be in sync with baselines

### PR DESCRIPTION
**Pull request by krishna5891**
Closes #373

### CODE REVIEW

**Comment by krishna5891** [comment]
```
krishna@turbotindias-MacBook-Pro aws_check_stack % terraform apply
var.turbot_profile
  Enter profile matching your turbot cli credentials.

  Enter a value: punisher


An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # turbot_policy_setting.aws_account_iam_stack will be created
  + resource "turbot_policy_setting" "aws_account_iam_stack" {
      + id                           = (known after apply)
      + precedence                   = "REQUIRED"
      + resource                     = (known after apply)
      + resource_akas                = (known after apply)
      + type                         = "tmod:@turbot/aws#/policy/types/accountStack"
      + value                        = "Check: Configured"
      + value_key_fingerprint        = (known after apply)
      + value_source                 = (known after apply)
      + value_source_key_fingerprint = (known after apply)
      + value_source_used            = (known after apply)
    }

  # turbot_policy_setting.aws_account_iam_stack_source will be created
  + resource "turbot_policy_setting" "aws_account_iam_stack_source" {
      + id                           = (known after apply)
      + precedence                   = "REQUIRED"
      + resource                     = (known after apply)
      + resource_akas                = (known after apply)
      + type                         = "tmod:@turbot/aws#/policy/types/accountStackSource"
      + value                        = <<~EOT
            resource "aws_iam_role" "example_role" {
              name               = "example_role"
              assume_role_policy = <<-EOF
                {
                  "Version": "2012-10-17",
                  "Statement": [
                    {
                      "Effect": "Allow",
                      "Principal": {
                        "AWS": "arn:aws:iam::287590803701:root"
                      },
                      "Action": "sts:AssumeRole",
                      "Condition": {
                        "StringEquals": {
                          "sts:ExternalId": "12345678"
                        }
                      }
                    }
                  ]
                }
                EOF
            }
            
            resource "aws_iam_policy" "example_policy" {
              name   = "example_policy"
              policy = <<-EOF
                {
                  "Version": "2012-10-17",
                  "Statement": [
                    {
                      "Action": [
                        "logs:CreateLogGroup",
                        "logs:CreateLogStream",
                        "logs:DescribeLogGroups",
                        "logs:DescribeLogStreams"
                      ],
                      "Effect": "Allow",
                      "Resource": "*"
                    }
                  ]
                }
                EOF
            }
            
            resource "aws_iam_role_policy_attachment" "example_role_policy_attach" {
              role       = aws_iam_role.example_role.name
              policy_arn = aws_iam_policy.example_policy.arn
            }
            
        EOT
      + value_key_fingerprint        = (known after apply)
      + value_source                 = (known after apply)
      + value_source_key_fingerprint = (known after apply)
      + value_source_used            = (known after apply)
    }

  # turbot_policy_setting.aws_account_iam_stack_tfversion will be created
  + resource "turbot_policy_setting" "aws_account_iam_stack_tfversion" {
      + id                           = (known after apply)
      + precedence                   = "REQUIRED"
      + resource                     = (known after apply)
      + resource_akas                = (known after apply)
      + type                         = "tmod:@turbot/aws#/policy/types/accountStackTerraformVersion"
      + value                        = "0.13.*"
      + value_key_fingerprint        = (known after apply)
      + value_source                 = (known after apply)
      + value_source_key_fingerprint = (known after apply)
      + value_source_used            = (known after apply)
    }

  # turbot_policy_setting.aws_iam_permissions_custom_levels_account will be created
  + resource "turbot_policy_setting" "aws_iam_permissions_custom_levels_account" {
      + id                           = (known after apply)
      + precedence                   = "REQUIRED"
      + resource                     = (known after apply)
      + resource_akas                = (known after apply)
      + type                         = "tmod:@turbot/aws-iam#/policy/types/permissionsCustomLevelsAccount"
      + value                        = <<~EOT
            - example-role
        EOT
      + value_key_fingerprint        = (known after apply)
      + value_source                 = (known after apply)
      + value_source_key_fingerprint = (known after apply)
      + value_source_used            = (known after apply)
    }

  # turbot_smart_folder.aws_stack will be created
  + resource "turbot_smart_folder" "aws_stack" {
      + description = "Defines sets of policies for the AWS Stack baseline"
      + id          = (known after apply)
      + parent      = "tmod:@turbot/turbot#/"
      + parent_akas = (known after apply)
      + title       = "AWS Stack Example Policies"
    }

Plan: 5 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

turbot_smart_folder.aws_stack: Creating...
turbot_smart_folder.aws_stack: Creation complete after 2s [id=215895555525855]
turbot_policy_setting.aws_iam_permissions_custom_levels_account: Creating...
turbot_policy_setting.aws_account_iam_stack: Creating...
turbot_policy_setting.aws_account_iam_stack_source: Creating...
turbot_policy_setting.aws_account_iam_stack_tfversion: Creating...
turbot_policy_setting.aws_account_iam_stack_tfversion: Creation complete after 2s [id=215895558046948]
turbot_policy_setting.aws_account_iam_stack: Creation complete after 8s [id=215895564172523]
turbot_policy_setting.aws_iam_permissions_custom_levels_account: Creation complete after 8s [id=215895564274927]
turbot_policy_setting.aws_account_iam_stack_source: Creation complete after 8s [id=215895564231917]

Apply complete! Resources: 5 added, 0 changed, 0 destroyed.

Outputs:

smart_folder_description = Defines sets of policies for the AWS Stack baseline
smart_folder_name = AWS Stack Example Policies
smart_folder_parent_resource = tmod:@turbot/turbot#/
turbot_profile = punisher
krishna@turbotindias-MacBook-Pro aws_check_stack %

**Comment by krishna5891** [comment]
Screenshot 2021-02-09 at 3 56 28 PM

File: baselines/getting_started/aws/aws_check_stack/README.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+# Baseline - AWS Check Stack
+
+AAWS Check Stack will allow you to set policy to deploy AWS Account IAM Stack , Terraform version for you source and also to apply the Stack Source policy.
```
Comment (omerosaienni) [inline baselines/getting_started/aws/aws_check_stack/README.md:3]: AAWS

File: baselines/getting_started/aws/aws_check_stack/aws_account_iam_stack.tf [hunk 2]
```diff
@@ -1,19 +1,22 @@
-## Set policy to deploy example AWS Account IAM Stack
+# AWS > Account > Stack
+# https://turbot.com/v5/mods/turbot/aws/inspect#/policy/types/accountStack
 resource "turbot_policy_setting" "aws_account_iam_stack" {
   resource = turbot_smart_folder.aws_stack.id
   type     = "tmod:@turbot/aws#/policy/types/accountStack"
   value    = "Check: Configured"
   #value    = "Enforce: Configured"
```
Comment (omerosaienni) [inline baselines/getting_started/aws/aws_check_stack/aws_account_iam_stack.tf:7]: remove

File: baselines/getting_started/aws/aws_check_stack/aws_account_iam_stack.tf [hunk 3]
```diff
@@ -22,10 +25,8 @@ resource "turbot_policy_setting" "aws_account_iam_stack_source" {
     SOURCE
```
Comment (omerosaienni) [inline baselines/getting_started/aws/aws_check_stack/aws_account_iam_stack.tf:25]: this one looks like it cannot run, did you try to terraform apply this?

File: baselines/getting_started/aws/aws_check_stack/aws_account_iam_stack.tf [hunk 4]
```diff
@@ -22,10 +25,8 @@ resource "turbot_policy_setting" "aws_account_iam_stack_source" {
...
 }
 
-# This is part of the Turbot RBAC use case if you are creating an IAM Role through Turbot
```
Comment (omerosaienni) [inline baselines/getting_started/aws/aws_check_stack/aws_account_iam_stack.tf:25]: Do we have this comment anywhere?

**Review by omerosaienni — CHANGES_REQUESTED** [review]

**Comment by omerosaienni** [comment]
All files that have policies should have the filename *_policies.tf

Can you do this?

File: baselines/getting_started/aws/aws_check_stack/aws_account_iam_stack.tf (hunk 3 above)
Comment (krishna5891) [reply baselines/getting_started/aws/aws_check_stack/aws_account_iam_stack.tf:25]: Yes this was tested before the commit and the output/screenshots were attached in this PR aswell.

File: baselines/getting_started/aws/aws_check_stack/aws_account_iam_stack.tf (hunk 4 above)
Comment (krishna5891) [reply baselines/getting_started/aws/aws_check_stack/aws_account_iam_stack.tf:25]: The same is available in README.me

File: baselines/getting_started/aws/aws_check_stack/aws_account_iam_stack.tf (hunk 2 above)
Comment (krishna5891) [reply baselines/getting_started/aws/aws_check_stack/aws_account_iam_stack.tf:7]: not sure what should need to be removed here.

File: baselines/getting_started/aws/aws_check_stack/README.md (hunk 1 above)
Comment (krishna5891) [reply baselines/getting_started/aws/aws_check_stack/README.md]: Typo corrected.

**Comment by krishna5891** [comment]
renamed the tf files that contains policies <*_policies.tf> format

**Pull request merged**